lib/clusterManager.js

```diff
@@ -521,6 +583,88 @@ if (cluster.isMaster) {
     );
   }
 
+  ClusterManager.prototype.getStorage = function() {
+    return this.callClusterMethodRemote(null, "clusterManager", "getStorage", [],
+      function(resultHandler) { 
+        return resultHandler[0]; // Return the first arg (storage object) of getStorage
+      },
+      function(e) {
+        clusterLogger.warn("Unable to retrieve storage object from cluster", e);
+      }, 5000 /* We need to give ample timeout time in case we attempt to retrieve storage
+      before the cluster is done making it. */
+    ).then((storage) => {
+      return storage;
+    }).catch((e) => {
+      clusterLogger.warn("Unable to retrieve storage data from cluster promise", e);
+      return e;
+    })
+  }
+
+  /* Though set here can be called like sync, we provide a callback in case we need
+  to wait for a set response */
+  ClusterManager.prototype.setStorageValue = function(id, value) {
+    return this.callClusterMethodRemote(null, "clusterManager", "setStorageValue", [id, value],
+      function(resultHandler) {
+        return resultHandler[0];
+      },
+      function(e) {
+        clusterLogger.warn("Error updating the storage: " + e);
+      }
+    );
+  }
+
+  /* Though set here can be called like sync, we provide a callback in case we need
+  to wait for a set response */
+  ClusterManager.prototype.setStorage = function(dict) {
+    return this.callClusterMethodRemote(null, "clusterManager", "setStorage", [dict],
+      function(resultHandler) {
+        return resultHandler[0];
+      },
+      function(e) {
+        clusterLogger.warn("Error updating the storage: " + e);
+      }
+    );
+  }
+
+  /* Though add here can be called like sync, we provide a callback in case we need
+  to wait for an add response */
+  ClusterManager.prototype.addStorage = function(dict) {
+    return this.callClusterMethodRemote(null, "clusterManager", "addStorage", [dict],
+      function(resultHandler) {
+        return resultHandler[0];
+      },
+      function(e) {
+        clusterLogger.warn("Error adding the storage: " + e);
+      }
+    );
+  }
+
+  ClusterManager.prototype.getStorageValue = function(id) {
+    return this.callClusterMethodRemote(null, "clusterManager", "getStorageValue", [id],
+    function(resultHandler) { 
+      return resultHandler[0]; // Return the first arg (storage object) of getStorage
+    },
+    function(e) {
+      clusterLogger.warn("Unable to retrieve storage object from cluster", e);
```

Comment:
Should be 'unable to retrieve get storage value'

Reply:
Thanks, good catch.